### PR TITLE
Add VCR, Faraday

### PR DESCRIPTION
**Pull request by be9**
1. Add pre-recorded API responses for blazing fast test runs!
2. Use Faraday instead of plain `Net::HTTP` (incorporating #173).

### CODE REVIEW

**Comment by be9** [comment]
@chrisb What versions of faraday and faraday_middleware do you use?

I want to add constraints to the dependencies' versions. Will 

```
  spec.add_dependency 'faraday', '>= 0.12'
  spec.add_dependency 'faraday_middleware', '>= 0.11'
```
work for you?

**Comment by chrisb** [comment]
@be9 that should work! Thanks!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -4,6 +4,54 @@ After checking out the repo, run `bin/setup` to install dependencies. You can al
...
+
+export PLAID_RUBY_CLIENT_ID=the_real_client_id
+export PLAID_RUBY_SECRET=the_real_secret
+export PLAID_RUBY_PUBLIC_KEY=the_real_public_key
+
+exec "$@"
+```
+
+Make this file executable (`chmod +x tmp/env.sh`), and you'll be able to run tests with
+`tmp/env.sh rake test`.
```
Comment (chrisb) [inline CONTRIBUTING.md]: Hi @be9! May I suggest using the [Dotenv gem](https://github.com/bkeepers/dotenv) for this purpose, instead? Within the repo you could supply a `.env.sample` (and add `.env` to `.gitignore`).

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -4,6 +4,54 @@ After checking out the repo, run `bin/setup` to install dependencies. You can al
...
+
+Another mode employs pre-recorded API responses using the
+[vcr](https://github.com/vcr/vcr) gem. It runs much faster. Just use
+`STUB_API=1 rake test` and you're good to go even without `env.sh`!
```
Comment (chrisb) [inline CONTRIBUTING.md]: I think this should be the default, no?

**Pull request merged**